src/parser.rs

```diff
@@ -71,6 +73,35 @@ fn parse_union_array_index(matcher_rule: pest::iterators::Pair<Rule>) -> UnionEl
     UnionElement::Index(i)
 }
 
+fn parse_union_array_slice(matcher_rule: pest::iterators::Pair<Rule>) -> UnionElement {
+    let mut start: Option<isize> = None;
+    let mut end: Option<isize> = None;
+    let mut step: Option<isize> = None;
+    for r in matcher_rule.into_inner() {
+        match r.as_rule() {
+            Rule::sliceStart => {
+                start = Some(r.as_str().parse().unwrap());
+            }
+
+            Rule::sliceEnd => {
+                end = Some(r.as_str().parse().unwrap());
+            }
+
+            Rule::sliceStep => {
+                step = Some(r.as_str().parse().unwrap());
+            }
+
+            _ => panic!("invalid parse tree {:?}", r),
+        }
+    }
+
+    UnionElement::Slice(Slice {
+        start: start.map(Index::from).unwrap_or_default(),
+        end: end.map(Index::from).unwrap_or_default(),
```

Comment:
Ah, I see now why you wanted to add the `Default` trait impl into slyce. It was not strictly necessary (although it doesn't hurt), since the `From` trait implementations also come with a blanket impl of `Into`:

```suggestion
        start: start.into(),
        end: end.into(),
```

the slyce Readme shows how `into()` can be used, but it's a bit artificial in one-liners because for type
inference to work seamlessly you need variables with real types that force the hand (like in this case `start`, ...)